### PR TITLE
Export Props type alias for each component

### PR DESCRIPTION
**Pull request by sentience**
Users reported having to alias or write out the full type definition in consuming code, which was painful.

Before:

```elm
module Components.MyComponent exposing
    ( view
    )


view :
    { propA: String
    , propB: Maybe Bool
    , slot: Maybe String
    }
    -> Html msg
view props =
    …
```

After:
```elm
module Components.MyComponent exposing
    ( view
    , Props
    )


type alias Props =
    { propA: String
    , propB: Maybe Bool
    , slot: Maybe String
    }


view :
    Props
    -> Html msg
view props =
    …
```

This also introduces an initial integration test for the plugin, to cover this new functionality.

### CODE REVIEW

File: src/__tests__/output-elm.spec.ts
Comment (ActuallyACat) [inline src/__tests__/output-elm.spec.ts:238]: I'm getting a type error in each test, it seems like this object is missing `htmlParts: [];` 
https://github.com/ionic-team/stencil/blob/2af380f7e5f403ff71998ba6a34ac386e4468cf7/src/declarations/stencil-private.ts#L699

File: src/__tests__/output-elm.spec.ts
Comment (ActuallyACat) [inline src/__tests__/output-elm.spec.ts:13]: Nitpicking - but somewhere in this block would be a good place to put a `toMatchSnapshot` assertion. This is the kind of scenario where that function is most useful.

**Review by ActuallyACat — CHANGES_REQUESTED** [review]

File: src/__tests__/output-elm.spec.ts
Comment (sentience) [reply src/__tests__/output-elm.spec.ts:13]: Disagree, I think. I don't like snapshot tests where the test does not describe the behaviour it is intended to test. These make it very hard for future maintainers to know if a failing test is a problem or not, and whether it's safe to generate a new snapshot.

I much prefer my approach here that tests the parts of the output that I care about, and makes tests that describe the intended behaviour of the system.

File: src/__tests__/output-elm.spec.ts
Comment (sentience) [reply src/__tests__/output-elm.spec.ts:238]: Weird, I'm not getting that error. I wonder if `htmlParts` was introduced in a newer version of Stencil than the one I'm testing with…

Comment (sentience) [reply src/__tests__/output-elm.spec.ts:238]: @ActuallyACat Did you happen to use Yarn to install the dependencies? The lock file is for NPM.

**Comment by sentience** [comment]
Just noticed something missing from @ActuallyACat's suggested output: type variables.

When a Props type contains an event handler:

```elm
    { label : String
    , slot : Maybe String
    , onAction : Maybe msg
    }
```

…its type alias must be declared with the `msg` type variable:

```elm
type alias Props msg =
    { label : String
    , slot : Maybe String
    , onAction : Maybe msg
    }
```

But when the Props type doesn't contain any events, it must _not_ declare a type variable:

```elm
type alias Props =
    { label : String
    , slot : Maybe String
    }
```

Get either of these wrong, and then Elm compiler complains.

This will be an easy addition, but I'll need to make it before we can merge this.

File: src/__tests__/output-elm.spec.ts
Comment (ActuallyACat) [reply src/__tests__/output-elm.spec.ts:238]: Oh that might be it

File: src/__tests__/output-elm.spec.ts
Comment (ActuallyACat) [reply src/__tests__/output-elm.spec.ts:13]: I definitely agree that snapshot tests can be noise that isn't always the best indicator of issues. Unintended side-effects would be the kinds of issues I'd expect snapshots to catch - e.g. maybe a change adds another attribute to `Props` but it's not reflected in view. 

Personally I would throw one in here, but I understand why you would not (hence the /nit :D)

**Review by ActuallyACat — APPROVED** [review]

**Pull request merged**
